plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -579,16 +584,90 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
     }
 
-    protected static String likeToRegexp(String like)
+    protected static boolean isSupportedLikeCall(Call call)
+    {
+        if (!LIKE_PATTERN_FUNCTION_NAME.equals(call.getFunctionName())) {
+            return false;
+        }
+
+        List<ConnectorExpression> arguments = call.getArguments();
+        if (arguments.size() < 2 || arguments.size() > 3) {
+            return false;
+        }
+
+        if (!(arguments.get(0) instanceof Variable) || !(arguments.get(1) instanceof Constant)) {
+            return false;
+        }
+
+        if (arguments.size() == 3) {
+            return arguments.get(2) instanceof Constant;
+        }
+
+        return true;
+    }
+
+    protected static String likeToRegexp(Slice pattern, Optional<Slice> escape)
+    {
+        char escapeChar = escape.map(ElasticsearchMetadata::getEscapeChar)
+                //
+                .orElse('\0');
+        StringBuilder regex = new StringBuilder();
+        boolean escaped = false;
+        int position = 0;
+        while (position < pattern.length()) {
+            int currentChar = getCodePointAt(pattern, position);
+            position += 1;
+            checkEscape(!escaped || currentChar == '%' || currentChar == '_' || currentChar == escapeChar);
+            if (!escaped && (currentChar == escapeChar)) {
+                escaped = true;
+            }
+            else {
+                switch (currentChar) {
+                    case '%':
+                        regex.append(escaped ? "%" : ".*");
+                        escaped = false;
+                        break;
+                    case '_':
+                        regex.append(escaped ? "_" : ".");
+                        escaped = false;
+                        break;
+                    case '\\':
+                        regex.append("\\\\");
+                        break;
+                    default:
+                        // escape special regex characters
+                        char[] chars = Character.toChars(currentChar);
+                        if (chars.length == 1 && REGEXP_RESERVED_CHARACTERS.contains(chars[0])) {
+                            regex.append('\\');
+                        }
```

Comment:
This looks somewhat overhead-ish. Why not just make REGEXP_RESERVED_CHARACTERS a set of code points?